tests/planner/ompl/test_TrajectoryConversions.cpp

```diff
@@ -0,0 +1,194 @@
+#include "eigen_tests.hpp"
+
+static const double eigenTolerance{1e-6};
+
+using StateSpace = aikido::statespace::dart::MetaSkeletonStateSpace;
+using aikido::planner::ompl::getSpaceInformation;
+
+
+namespace {
+
+  aikido::trajectory::InterpolatedPtr constructTrajectory(
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
+      aikido::statespace::InterpolatorPtr _interpolator,
+      const Eigen::Vector3d& _startPose,
+      const Eigen::Vector3d& _midwayPose, 
+      const Eigen::Vector3d& _goalPose          
+    )
+  {
+
+    auto startState = _stateSpace->createState();
+    auto subState1 = _stateSpace->getSubStateHandle<R3>(startState, 0);
+    subState1.setValue(_startPose);
+
+    auto midwayState = _stateSpace->createState();
+    auto subState2 = _stateSpace->getSubStateHandle<R3>(midwayState, 0);
+    subState2.setValue(_midwayPose);
+
+    auto goalState = _stateSpace->createState();
+    auto subState3 = _stateSpace->getSubStateHandle<R3>(goalState, 0);
+    subState3.setValue(_goalPose);
+
+    // Construct Trajectory
+    auto returnInterpolated = std::make_shared<aikido::trajectory::Interpolated>(
+        std::move(_stateSpace), std::move(_interpolator));
+
+    returnInterpolated->addWaypoint(0, startState);
+    returnInterpolated->addWaypoint(1, midwayState);
+    returnInterpolated->addWaypoint(2, goalState);
+
+    return returnInterpolated;
+  }
+
+}
+
+// Test if the durations of the two trajectories remains constant
+TEST_F(PlannerTest, OMPLDurationRemainsUnchanged)
+{
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d midwayPose(0, -2, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  // Construct a test trajectory
+  auto traj = constructTrajectory(stateSpace, interpolator, startPose, midwayPose, goalPose);
+
+  // Get the ompl state space
+  auto si = getSpaceInformation(
+      stateSpace,
+      interpolator,
+      std::move(dmetric),
+      std::move(sampler),
+      std::move(collConstraint),
+      std::move(boundsConstraint),
+      std::move(boundsProjection),
+      0.1);
+
+  auto omplTraj = aikido::planner::ompl::toOMPLTrajectory(
+      traj, si);
+
+  EXPECT_TRUE(omplTraj.getStateCount() == traj->getNumWaypoints());
+
+}
+
+// Tests that toOMPLTrajectory function does not alter the waypoints 
+TEST_F(PlannerTest, OMPLStatesRemainUnchaged)
+{
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d midwayPose(0, -2, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  // Construct a test trajectory
+  auto traj = constructTrajectory(stateSpace, interpolator, startPose, midwayPose, goalPose);
+
+  // Get the ompl state space
+  auto si = getSpaceInformation(
+      stateSpace,
+      interpolator,
+      std::move(dmetric),
+      std::move(sampler),
+      std::move(collConstraint),
+      std::move(boundsConstraint),
+      std::move(boundsProjection),
+      0.1);
+
+  auto omplTraj = aikido::planner::ompl::toOMPLTrajectory(
+      traj, si);
+  
+  // Match every point
```

Comment:
Nope:

1. All tests are independent. GTest will continue to run tests even after one tests fail. This is useful because it often uncovers multiple bugs at once in one Travis run.
2. Tests aren't guaranteed to be ordered in any particular way. Or even that all of them are run - you can run a subset of tests by passing commandline arguments to GTest.
3. The `EXPECT_` macros intentionally allow testcase to continue and mark it as failed once it is done running. Only `ASSERT_` macros cause the test to immediately exit.

In general, all tests should be independent and a test should never crash, only gracefully report failure. If a check is critical to avoid a SEGFAULT (e.g. a length check like this), you should use `ASSERT_` instead of `EXPECT_`.